### PR TITLE
Skeleton tool : Conditional disable/enable of Deformation layer on tool switch

### PR DESCRIPTION
**Pull request by Aa20475**
Issue #1624

### CODE REVIEW

File: synfig-studio/src/gui/states/state_bone.cpp [hunk 1]
```diff
@@ -437,6 +437,9 @@ StateBone_Context::StateBone_Context(CanvasView *canvas_view) :
 	if(etl::handle<Layer_SkeletonDeformation>::cast_dynamic(layer)){
 		get_work_area()->set_type_mask(get_work_area()->get_type_mask()-Duck::TYPE_TANGENT|Duck::TYPE_WIDTH);
 		get_canvas_view()->toggle_duck_mask(Duck::TYPE_NONE);
+		cout<<"Hi"<<endl;
```
Comment (rodolforg) [inline synfig-studio/src/gui/states/state_bone.cpp:440]: ops ;)

File: synfig-studio/src/gui/states/state_bone.cpp [hunk 2]
```diff
@@ -437,6 +437,9 @@ StateBone_Context::StateBone_Context(CanvasView *canvas_view) :
...
+		layer->disable();
+		get_canvas_interface()->signal_layer_status_changed()(layer,false);
```
Comment (rodolforg) [inline synfig-studio/src/gui/states/state_bone.cpp:442]: it should not be necessary at all!
`void disable() { set_active(false); }` calls `set_active()`

And `set_active()` calls the signal
```c++
Layer::set_active(bool x)
{
	if(active_!=x)
	{
		active_=x;

		Node::on_changed();
		signal_status_changed_();
	}
```

Comment (Aa20475) [reply synfig-studio/src/gui/states/state_bone.cpp:442]: Hey! For some reason, without the  `get_canvas_interface()->signal_layer_status_changed()(layer,false);` , the `dock_layers` is not being updated. :thinking:

**Comment by morevnaproject** [comment]
It works well. Just make sure to remove redundant comment. ^__^

File: synfig-studio/src/gui/states/state_bone.cpp (hunk 2 above)
Comment (rodolforg) [reply synfig-studio/src/gui/states/state_bone.cpp:441]: Ah I got it.
The layer panel doesn't monitor it indeed. And it makes sense somehow (due to switching between open documents) :)
So the solution is the way you did or calling an synfigapp action as Konstantin pointed.

**Comment by morevnaproject** [comment]
Merged now. Thank you!

**Pull request merged**